### ISSUE TITLE
Missing global:: prefix in generated code

### ISSUE THREAD

**Issue by Liversage** [dotnet/orleans#5244]
I'm using `Microsoft.Orleans.CodeGenerator.MSBuild` and I got this compile error:

> The type or namespace name 'Reflection' does not exist in the namespace 'MyNamespace.System' (are you missing an assembly reference?)

Looking at the generated code `MyProject.orleans.gs` I noticed the generator had created code like this:

```
global::System.Reflection.FieldInfo field0 = typeof(global::MyNamespace.MyClass).GetField("<AnotherClass>k__BackingField", (System.Reflection.BindingFlags.Instance | System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Public));
```

My project has a `MyNamespace.System` namespace so `System.Reflection.BindingFlags` tries to resolve the namespace `MyNamespace.System.Reflection` namespace which of course doesn't exist.

The problem is that `System.Reflection.BindingFlags` in the generated code is missing the `global::` prefix.

**Comment by ReubenBond** [comment]
Hi Martin, thanks for reporting. Are you able to submit a PR to fix this?

**Comment by Liversage** [comment]
@ReubenBond I would be happy to submit a PR but right now I'm knee deep into issues with my own project so it might take some time before I will be able to do it.

**Comment by Liversage** [comment]
After upgrading from 2.1.2 to 2.2.0 I can no longer reproduce the issue. However, I still don't understand what went wrong in 2.1.2. My issue is actually a duplicate of  #5067 where it was fixed in the new (MSBuild) code generator. However (and to my confusion) the two projects `Orleans.CodeGeneration` (old) and `Orleans.CodeGenerator` (new) contain a lot of duplicated code.

I believe my specific problem was caused by [this line of code](https://github.com/dotnet/orleans/blob/master/src/Orleans.CodeGeneration/Utilities/SyntaxFactoryExtensions.cs#L115) in `Orleans.CodeGeneration`:

    var flags = SyntaxFactory.IdentifierName("System").Member("Reflection").Member("BindingFlags");

In `Orleans.CodeGenerator` it became [this line of code](https://github.com/dotnet/orleans/blob/master/src/Orleans.CodeGenerator/Utilities/SymbolSyntaxExtensions.cs#L119) that was introduced when #5067 was fixed:

    var flags = AliasQualifiedName("global", IdentifierName("System")).Member("Reflection").Member("BindingFlags");

I still cannot explain why switching to the new MSBuild code generator in 2.1.2 resulted in this issue because it only exists in the old code generator but now that 2.2.0 is out I don't think it is worth the effort to investigate.

You can argue that it is better to also fix the old code generator but from what I have seen the missing `global::` prefix was not a problem because the generated code without the prefix only appears in an Orleans generated namespace and not inside the namespace of my project.

I'm not convinced that my analysis is entirely correct so I would prefer to not make a pull request for one line of code to fix a rather special problem in an old version of Orleans. Instead, I suggest that this issue is closed.

**Comment by ReubenBond** [comment]
Ah, looks like I already fixed it 😉 but I opened #5269 to back-port the fix anyhow, just in case the old code generator continues to live for some time.
